store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -1883,6 +1884,10 @@ public void searchDirectory(SearchDirectoryOptions options, NamedEntry.Visitor v
          */
         Domain domain = options.getDomain();
         String[] bases = getSearchBases(domain, types);
+        if (options.getTypes().contains(ObjectType.habgroups)) {
```

Comment:
better to check the type before executing getSearchBases() as it's not required when type is HAB.

Reply:
Actually it is required.